src/Illuminate/Support/Str.php

```diff
@@ -1778,6 +1778,24 @@ public static function wordWrap($string, $characters = 75, $break = "\n", $cutLo
         return wordwrap($string, $characters, $break, $cutLongWords);
     }
 
+    /**
+     * Counts the number of lines in the input string.
+     *
+     * @param  string  $string
+     * @param  bool  $withEmptyLines
+     * @return int
+     */
+    public static function lineCount(string $string, bool $withEmptyLines = true): int
+    {
+        $string = str_replace("\r\n", "\n", $string);
+
+        if ($withEmptyLines) {
+            return count(explode("\n", $string));
+        }
+
+        return count(array_filter(explode("\n", $string), 'mb_strlen'));
```

Comment:
You can also try to avoid this for non-empty lines only:
```suggestion
        $lines = preg_match_all(
            '/(?(DEFINE)(?<lineBreak>\r\n|\r|\n))(.+)(?P>lineBreak)/',
            $string
        );
        return $lines === 0 && $string !== '' ? 1 : $lines;
```